pkg/operator/ceph/controller/spec.go

```diff
@@ -615,6 +634,56 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 	}
 }
 
+// GenerateMDSLivenessProbeExecDaemon generates a liveness probe that makes sure mds daemon is present in fs map,
+// that it can be called, and that it returns 0
+func GenerateMDSLivenessProbeExecDaemon(daemonID, filesystemName string) *v1.Probe {
+	var mdsLivenessProbeCmd = []string{
+		"timeout",
+		fmt.Sprintf("%d", mdsCmdTimeout),
+		"sh",
+		"-c",
+		fmt.Sprintf(`
+outp="$(ceph fs dump --mon-host=$(ROOK_CEPH_MON_HOST) --mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS) --format json 2>&1)"
+rc=$?
+if [ $rc -ne 0 ]; then
+	echo "ceph mds dump check failed with the following output:"
+	echo "$outp" | sed -e 's/^/> /g'
+	echo "we couldn't get the information to reliably tell if the MDS should be restarted or not"
+else
+	# get the active and standby mds in the fs map
+	standbyMds=$(echo $outp | jq ".standbys|map(.name)")
+	activeMds=$(ceph fs get %s --mon-host=$(ROOK_CEPH_MON_HOST) --mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS) --format json 2>&1 | jq ".mdsmap.info|map(.name)")
+	rc1=$?
+	if [ $rc1 -ne 0 ]; then
+		echo "ceph mds get check failed with the following output:"
+		echo "$activeMds" | sed -e 's/^/> /g'
+		exit
+	fi
+	daemonMds=%s
+	if [[ ${activeMds[*]} =~ $daemonMds || ${standbyMds[*]} =~ $daemonMds ]]; then
+		echo "filesystem present in mds map, no need to re-start the container"
+	else
+		echo "filesystem not present in mds map, no need to re-start the container"
+		exit
+	fi
+fi
```

Comment:
I have found that when a script gets past like 8 lines, it's often easier to dev/test when the script is a separate file that is imported using `//go:embed`. It also allows running shellcheck on the file which can help catch some errors.

I'd suggest making a `mds-liveness.sh` script in the `pkg/operator/ceph/file/mds` directory and importing it there.

Reply:
You can look at pkg/daemon/multus/templates.go for what I did recently for templating yaml files, which is pretty similar.

Reply:
Thanks updated